app/uk/gov/hmrc/gatekeeperemail/services/DraftEmailService.scala

```diff
@@ -39,15 +41,14 @@ class DraftEmailService @Inject() (
     apmConnector: ApmConnector,
     draftEmailRepository: DraftEmailRepository,
     sentEmailRepository: SentEmailRepository,
-    appConfig: AppConfig
+    appConfig: AppConfig,
+    clock: Clock
   )(implicit val ec: ExecutionContext
   ) {
 
   val logger: Logger = Logger(getClass.getName)
 
   def persistEmail(emailRequest: EmailRequest, emailUUID: String): Future[DraftEmail] = {
```

Comment:
We could put a UUID binder in and change routes and endpoints to validate it before we accept it.